x-pack/plugins/security_solution/public/detections/hooks/trigger_actions_alert_table/use_cell_actions.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiDataGridColumn, EuiDataGridRefProps } from '@elastic/eui';
+import type { TimelineNonEcsData } from '@kbn/timelines-plugin/common';
+import { get } from 'lodash';
+import type { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import type { AlertsTableConfigurationRegistry } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { useCallback, useMemo } from 'react';
+import { tableDefaults } from '../../../common/store/data_table/defaults';
+import { VIEW_SELECTION } from '../../../../common/constants';
+import { useSourcererDataView } from '../../../common/containers/sourcerer';
+import { defaultCellActions } from '../../../common/lib/cell_actions/default_cell_actions';
+import type { TableId } from '../../../../common/types';
+import { FIELDS_WITHOUT_CELL_ACTIONS } from '../../../common/lib/cell_actions/constants';
+import { SourcererScopeName } from '../../../common/store/sourcerer/model';
+import { useShallowEqualSelector } from '../../../common/hooks/use_selector';
+import { dataTableSelectors } from '../../../common/store/data_table';
+
+export const getUseCellActionsHook = (
+  tableId: TableId
+): AlertsTableConfigurationRegistry['useCellActions'] => {
+  const useCellActions = ({
+    columns,
+    data,
+    ecsData,
+    dataGridRef,
+    pageSize,
+  }: {
+    // Hover Actions
+    columns: EuiDataGridColumn[];
+    data: unknown[][];
+    ecsData: unknown[];
+    dataGridRef?: EuiDataGridRefProps | null;
```

Comment:
Why is it called _dataGridRef_ it is not a ref? It is also causing many updates. 
I think trigger_actions_ui's `AlertsTable` should pass the actual ref (dataGridRef) to the hook, instead of the value (dataGridRef.current).

Reply:
You are right. Thanks for pointing it out. I have corrected it now and made changes to the [trigger actions plugin as well](https://github.com/elastic/kibana/pull/149128/files#diff-9fc5c6838da79440ff760a13ceed83c8b38811ec1454d60011c365a16eae302cR351-R356).